File: 1013-fibonacci-number/solution.py
import numpy as np
class Solution:
    def fib(self, N: int) -> int:
        if N <= 1:
            return N
        
        A = np.array(([1,1],[1,0]))
        A = np.linalg.matrix_power(A, N-1)
        return A[0][0]
        

File: 1013-fibonacci-number/solution.cpp
class Solution {
public:
    int fib(int n) {
        if (n == 0) {
            return 0;
        }
        int cur_fib = 1, prev_fib = 0;
        for (int iter = 2; iter <= n; ++iter) {
            int temp = cur_fib;
            cur_fib += prev_fib;
            prev_fib = temp;
        }
        return cur_fib;
    }
};
